### ISSUE TITLE
Distances between Clusters and Batch Corrected Expression Data

### ISSUE THREAD

**Issue by hkarakurt8742** [satijalab/seurat#1107]
Hello,
I am using Seurat to cluster a scRNA-Seq data. As known, with different values of resolution, clusters vary. Is there a way to define the clustering quality or a way to measure distances between clusters?

Also, after the CCA, the new Seurat object (data.integrated) does not contain batch corrected expression values. Is there a way to obtain this data?

Thank you in advance
Hamza

**Comment by andrewwbutler** [comment]
Hi,

If you want to define clustering quality based on distances, the silhouette score would be one thing to look into. The `cluster` R package implements a function to calculate this (see [here](https://stat.ethz.ch/R-manual/R-devel/library/cluster/html/silhouette.html)). 

In Seurat v2, we do not return a corrected expression matrix but our new integration method implemented in Seurat v3 does. Please see this [vignette](https://satijalab.org/seurat/pancreas_integration_label_transfer.html) for details and an example workflow.

**Comment by hkarakurt8742** [comment]
> Hi,
> 
> If you want to define clustering quality based on distances, the silhouette score would be one thing to look into. The `cluster` R package implements a function to calculate this (see [here](https://stat.ethz.ch/R-manual/R-devel/library/cluster/html/silhouette.html)).
> 
> In Seurat v2, we do not return a corrected expression matrix but our new integration method implemented in Seurat v3 does. Please see this [vignette](https://satijalab.org/seurat/pancreas_integration_label_transfer.html) for details and an example workflow.

Thank you for answer.
Silhoutte function requires a distance matrix as I know. Does the SNN matrix in Seurat object works for that application? Also, I will use some external clustering approaches. If the SNN matrix works for Silhouette plots, can I use the same SNN matrix for these algorithms?


Thank you in advance.

**Comment by andrewwbutler** [comment]
I would calculate distances for silhouette scores in the same space that you used to compute the SNN. Not all clustering algorithms are graph-based but for those that are, you could use the same SNN matrix.

**Comment by hkarakurt8742** [comment]
> I would calculate distances for silhouette scores in the same space that you used to compute the SNN. Not all clustering algorithms are graph-based but for those that are, you could use the same SNN matrix.

Hello again,
Sorry for so many questions. I tried the method as you said but my average plot score is -1. I think it is a really bad score. 
I tried it with the PBMC tutorial you provided. I am using the clusters and SNN matrix stored in Seurat object. It is the same. All scores are -1. 

I am not sure about the reason. If you have an idea, I would be so grateful.

Thanks.
